client/src/components/PlanShowPage.js

```diff
@@ -14,18 +17,83 @@ const PlanShowPage = (props) => {
     }
   }
 
+  const fetchUser = async () => {
+    const currentUser = await getCurrentUser()
+    setUser(currentUser)
+  }
+
   useEffect(() => {
-    fetchPlan()
+    fetchPlan(),
+      fetchUser()
   }, [])
 
-  const detailComponents = Object.keys(plan).map(attr => {
-    return <p>{_.capitalize(attr)}: {plan[attr]}</p>
-  })
+  const handleClick = async (event) => {
+    try {
+      const response = await fetch(`/api/v1/signups`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify({ planId: plan.id, userId: user.id })
```

Comment:
you dont actually have to add `userId` here. In the router, you can all `req.user` and it will provide that user to you. But its not terrible if you add it here. So dealers choice!